v1.16 Backports 2025-06-03

tklauser opened the pull request:
* [x] #37362 (@Artyop) :warning: resolved conflicts
 * [ ] #39802 (@aanm) :warning: resolved conflicts

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 37362 39802
```

aanm committed:
.github/workflows: remove clean up disk-space step

[ upstream commit 63be083d80a7c5f422ff1f82910880f770f78b3c ]

The runners have enough disk-space to build the images, we can remove
this step which is slowing down CI build time.

Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

aanm committed:
Revert "ci: stop using /mnt/ directory for building images"

[ upstream commit db7ed53906b9df86603071747ee039ba7ba428e6 ]

This reverts commit e90242efe6bca38959b20c4b27a7f93320c48bf2.

Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

aanm committed:
workflows/build-images-ci: debug disk usage in image build

[ upstream commit 56c985b580f1d20d596a09631ecb955979b320f6 ]

Remove temporary file in /mnt

In some runners, the file '/mnt/tmp-pv.img' is huge, taking the entire
space of '/mnt' directory. Thus, we will forcefully remove this file
which will allow us to re-use the /mnt directory again to build CI
images.

Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

Artyop commented on .github/workflows/tests-smoke.yaml:
```diff
@@ -183,9 +183,11 @@ jobs:
+          cilium_pod_ip=$(kubectl -n kube-system get po -o name --field-selector=status.phase==Running -l 'k8s-app=cilium' -o jsonpath='{.items[0].status.podIP}' )
+          curl http://${cilium_pod_ip}:9962/metrics > metrics-agent.prom
+
+          operator_ip=$(kubectl get pods -n kube-system -l io.cilium/app=operator -o jsonpath='{range .items[*]}{.status.podIP}{"\n"}{end}')
```
The part with operator_ip wasn't added in commit https://github.com/cilium/cilium/pull/37362/commits/4d85c9cc5a101f0d8a998302a847c41032741cee so I'm not sure it should be embedded here?

tklauser replied:
Yeah, looks like it was added in eabba03ad0b58966314cc2ab42bce93fe65f114e on `main` but that change wasn't backported to `v1.16`. I'll drop that bit from the backported commit.

Artyop committed:
ci: fix curl call for metrics in smoke tests

[ upstream commit db2feafc2320a7081713b36fae7183e34e0142f7 ]

This commit remove the usage of installing apt and curl inside the container by calling the metric endpoint from the runner instead.

Signed-off-by: Antony Reynaud <antony.reynaud@isovalent.com>
Signed-off-by: Tobias Klauser <tobias@cilium.io>
